MonkeyWrench.Web.WebService/WebServices.asmx.cs

```diff
@@ -3066,6 +3066,95 @@ FROM RevisionWork
 						}
 					}
 					break;
+
+				case DBQueueManagement.ChooseHighestPriorityLeastRecent:
+					if (hostlanes.Count > 1) {
+						int lane_index = -1;
+						int highest_priority = 0;
+						int p = 0;
+						DateTime latest_date = DateTime.MaxValue;
+						DateTime ld = DateTime.MaxValue;
+
+						// we need to find the latest revisionwork each hostlane has completed.
+						// we want to work on the hostlane which has waited the longest amount
+						// of time without getting work done (but which has pending work to do).
+
+						for (int i = 0; i < hostlanes.Count; i++) {
+							DBHostLane hl = hostlanes [i];
+							// check if this hostlane has pending work and get its priority.
+							// this would ideally be included in the query below, but I'm not sure
+							// how to do that while still distinguising the case where nothing has
+							// been done ever for a hostlane.
+							using (IDbCommand cmd = db.CreateCommand ()) {
+								cmd.CommandText = @"
+SELECT RevisionWork.priority
+FROM RevisionWork
+WHERE
+    RevisionWork.host_id = @host_id
+AND (RevisionWork.workhost_id = @workhost_id OR RevisionWork.workhost_id IS NULL)
+AND RevisionWork.completed = false
+AND RevisionWork.state <> 9 AND RevisionWork.state <> 10 AND RevisionWork.state <> 11
+AND lane_id = @lane_id
+LIMIT 1;
```

Comment:
I think you should order by `priority` here, so that you get the highest priority of all the pending revision work:

    ORDER BY priority DESC

This also means that when selecting the revision work to work on, you'll have to order by priority (in `DB.GetRevisionWork`).